### PR TITLE
Treat validation problems as warnings in HL API

### PR DESCRIPTION
**Pull request by qwepoizt**
~This PR is a case study for #4053.~

Improves #4053 and #4029 (don't autoclose!).

This PR implements:

* support for warnings in elektra_error.c
* report validation problems as warnings (instead of errors) in range plugin.
* treat warnings like errors in HL API.

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy. If not, fix **in this order**:
  - [x] add a line in `doc/news/_preparation_next_release.md`
  - [x] reformat the code with `scripts/dev/reformat-all`
  - [x] make all unit tests pass
  - [x] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation
- [x] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by qwepoizt** [comment]
@markus2330: As discussed in https://github.com/ElektraInitiative/libelektra/issues/4053#issuecomment-922901984 I've set up a PR that:

* changes the range plugin to report errors as warnings during kdbGet()
* transforms all warnings to errors in the HL API.

I like the results! E.g. for redshift the behavior is now:

```
$ ./redshift --temp-day 50
Opening redshift's configuration failed: 
Internal: Value '50' of key 'proc:/sw/jonls/redshift/#0/current/temp/day' not within range 1000-25000
```

I'd be happy to refactor this a bit and merge it.

What do you think?

**Comment by qwepoizt** [comment]
> @markus2330: As discussed in [#4053 (comment)](https://github.com/ElektraInitiative/libelektra/issues/4053#issuecomment-922901984) I've set up a PR that:
> 
>     * changes the range plugin to report errors as warnings during kdbGet()
> 
>     * transforms all warnings to errors in the HL API.
> 
> 
> I like the results! E.g. for redshift the behavior is now:
> 
> ```
> $ ./redshift --temp-day 50
> Opening redshift's configuration failed: 
> Internal: Value '50' of key 'proc:/sw/jonls/redshift/#0/current/temp/day' not within range 1000-25000
> ```
> 
> I'd be happy to refactor this a bit and merge it.
> 
> What do you think?

@markus2330 I've refactored and believe this is ready to merge!

File: src/libs/highlevel/elektra_error.c [hunk 1]
```diff
@@ -83,6 +83,8 @@ void elektraErrorAddWarning (ElektraError * error, ElektraError * warning)
  * If no error exists, a pure warning error will be used.
  * @see elektraErrorPureWarning
  *
+ * @note In most cases, errors and warnings were added to the key using src/libs/elektra/errors.c.
```
Comment (markus2330) [inline src/libs/highlevel/elektra_error.c:86]: What means "In most cases"?

**Review by markus2330 — APPROVED** [review]
Amazing work that you fixed that! Can you add a small example and/or test case how to retrieve warnings? Probably best in the tutorial.

File: src/libs/highlevel/elektra_error.c (hunk 1 above)
Comment (qwepoizt) [reply src/libs/highlevel/elektra_error.c:86]: Thanks, I'll update the comment to make it clearer.

**Comment by qwepoizt** [comment]
> Amazing work that you fixed that! Can you add a small example and/or test case how to retrieve warnings? Probably best in the tutorial.

Thank you! Yes, sure. Do you mean https://www.libelektra.org/tutorials/high-level-api?

**Comment by mpranj** [comment]
Could you rebase this to fix the checks?

**Comment by qwepoizt** [comment]
> Could you rebase this to fix the checks?

Yes, I did. Good idea!

**Comment by qwepoizt** [comment]
> > Amazing work that you fixed that! Can you add a small example and/or test case how to retrieve warnings? Probably best in the tutorial.
> 
> Thank you! Yes, sure. Do you mean https://www.libelektra.org/tutorials/high-level-api?

@mpranj I'll do this in a new PR, to not delay this one further.

**Comment by mpranj** [comment]
Thank you so much!

**Pull request merged**